### PR TITLE
JDK-8314883: Java_java_util_prefs_FileSystemPreferences_lockFile0 write result errno in missing case

### PR DESCRIPTION
**Pull request by MBaesken**
There seems to be a codepath in Java_java_util_prefs_FileSystemPreferences_lockFile0 where the errno is not stored but potentially accessed in the calling Java code.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314883](https://bugs.openjdk.org/browse/JDK-8314883): Java_java_util_prefs_FileSystemPreferences_lockFile0 write result errno in missing case (**Bug** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Vyom Tewari](https://openjdk.org/census#vtewari) (@vyommani - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15402/head:pull/15402` \
`$ git checkout pull/15402`

Update a local copy of the PR: \
`$ git checkout pull/15402` \
`$ git pull https://git.openjdk.org/jdk.git pull/15402/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15402`

View PR using the GUI difftool: \
`$ git pr show -t 15402`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15402.diff">https://git.openjdk.org/jdk/pull/15402.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15402#issuecomment-1690008810)

### CODE REVIEW

**Review by jaikiran — APPROVED** [review]
The change looks OK to me.

Please wait for another review from someone more familiar with this code.

**Comment by MBaesken** [comment]
Hi jaikiran, thanks for the review !

**Review by shipilev — APPROVED** [review]
Looks reasonable.

**Review by vyommani — APPROVED** [review]
Looks OK to me.

**Comment by MBaesken** [comment]
Thanks for the reviews !

/integrate

**Pull request closed without merging**
